model/jpa/src/main/java/org/keycloak/storage/jpa/JpaUserFederatedStorageProvider.java

```diff
@@ -165,6 +165,11 @@ public MultivaluedHashMap<String, String> getAttributes(RealmModel realm, String
 
     @Override
     public Stream<String> getUsersByUserAttributeStream(RealmModel realm, String name, String value) {
+        if (value.length() > 2024) {
+            //this should never happen as the exception should be thrown already by JpaUserProvider#getUsersByUserAttributeStream, but leaving it here as additional check
```

Comment:
Reading the comment, it looks like we previously supported searching for attributes longer than 255 characters in the federated attributes, and the code has changed to no longer support this. This would then be a regression. It is also difficult to explain to users why user attribute search lengths are different in both places.

Nitpick: I didn't find the method referenced here. It seems to me that the check is done in `searchForUserByUserAttributeStream`, therefore the comment might needs updating.

```suggestion
            //this should never happen as the exception should be thrown already by JpaUserProvider#searchForUserByUserAttributeStream, but leaving it here as additional check
```